Open Judge System/OJS.Services/Data/TestRuns/TestRunsDataService.cs

```diff
@@ -0,0 +1,19 @@
+﻿namespace OJS.Services.Data.TestRuns
+{
+    using OJS.Data.Models;
+    using OJS.Data.Repositories.Contracts;
+
+    public class TestRunsDataService : ITestRunsDataService
+    {
+        private readonly IEfGenericRepository<TestRun> testRuns;
+
+        public TestRunsDataService(IEfGenericRepository<TestRun> testRuns) =>
+            this.testRuns = testRuns;
+
+        public void DeleteByProblem(int problemId) =>
+            this.testRuns.Delete(tr => tr.Submission.ProblemId == problemId);
+
+        public void DeleteByContest(int contestId) =>
+            this.testRuns.Delete(tr => tr.Submission.Problem.ProblemGroup.ContestId == contestId);
```

Comment:
I might somewhat see a case for deleting all test runs for a problem (when retesting a problem), but deleting all test runs for a contest seems unnecessary, both problems and submissions are deletable entities so there is no need to delete anything in order to delete them.